### DISCUSSION TITLE
Fief and FastAPI-Users

### DISCUSSION THREAD

**Discussion by davidbrochart** [fief-dev/fief#60] [Q&A]
In Fief-Python, users have type `FiefUserInfo`, and in FastAPI-Users they have type `UserRead`. These types seem unrelated as we have:
```py
FiefUserInfo = Dict[str, Any]
```
But Fief seems to use FastAPI-Users under the hood, so I was wondering if somehow `FiefUserInfo` could be a subclass of `UserRead`, or is that completely impossible?
I have an application where I would like a user to be one of FastAPI-Users or Fief. Does is even make sense?

**Comment by frankie567** [comment]
Well, I would say that's not really the philosophy of Fief. I see two options for you:

* You use FastAPI Users for full customization and control about how you authenticate users and store their data
* You use Fief to provide you users identity; but you don't have direct access to the underlying data and you use its builtin authentication mechanism (which is an OAuth2/OpenID implementation).

**Comment by davidbrochart** [comment]
I see, and I guess it's not possible to "integrate" Fief in a FastAPI-Users `AuthenticationBackend`?
I was thinking that maybe we could register a new endpoint for Fief that would use a FastAPI-Users OAuth2 router.

**Reply by frankie567** [reply]
The most natural way would be to add it as an OAuth2 backend, like we can do for Google or GitHub. But IMO, we would lose most of Fief's benefits doing this way.

Could be interesting to have a call so we can talk about this and figure out the best solution for Jupyverse 🙂
